### PR TITLE
Make keys immutable

### LINKED ISSUE #1128: Minor redesign and optimization of (Private)Key classes
* Remove **_privateKey** field from **Key** base class.
* Remove `ctor(byte[] data)` from **Key**, and move current implementation to **RsaKey** and **DsaKey** classes.
* The `ctor(byte[] data` of **RsaKey** and **DsaKey** can be optimized to remove intermediate **List\<BigInteger>**.
* Remove initial assignment of **_publicKey** from `ED25519Key(byte[] pk, byte[] sk)`ctor.
* Rename **privateKey** field in **ED25519Key** to **_privateKey** and remove disable of **IDE1006** for this field.

### PR DESCRIPTION
**Pull request by Rob-Hague**
This makes it easier to reason about Key instances in e.g. DigitalSignature implementations, because we know that the Key is initialised with its data and will not change.

Also fixes #1128

### CODE REVIEW

File: test/Renci.SshNet.Benchmarks/Security/Cryptography/Ciphers/RsaCipherBenchmarks.cs
Comment (Rob-Hague) [inline test/Renci.SshNet.Benchmarks/Security/Cryptography/Ciphers/RsaCipherBenchmarks.cs:1]: I'll revisit this in a follow-up

**Review by WojciechNagorski — APPROVED** [review]
LGTM

**Review by WojciechNagorski — APPROVED** [review]
LGTM

**Comment by WojciechNagorski** [comment]
The 2023.0.1 version has been released to Nuget: https://www.nuget.org/packages/SSH.NET/2023.0.1

**Pull request merged**